taqueria-utils/taqueria-utils.ts

```diff
@@ -83,11 +84,29 @@ export const doesPathExist = (path: string) =>
 			})
 		));
 
+export const isPathEmptyDirectory = (path: string) =>
+	attemptP(async () => {
+		const dirInfo = Deno.readDir(path);
+		for await (const _item of dirInfo) {
```

Comment:
Is there another way to check that `dirInfo` is not an empty directory without using a for loop? This looks like a hack.

Reply:
I agree, @jchenche. Unfortunately, it Deno is using a relatively new ECMAScript feature called async iterators:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of

https://github.com/tc39/proposal-async-iteration

I'm not a fan of generators due to having intrinsic side-effects. It does seem that Kyle Simpson (author of Functional Light JavaScript book) wrote a library to handle these in a functional way:

https://github.com/getify/fasy

Nonetheless, as I'll need to take some time to assess that library I think rolling with what Ali has here is fine and the responsible approach.